### PR TITLE
Cancel allowance shouldn't delete contracts

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**
Fixes [#3013](https://github.com/NebulousLabs/Sia/issues/3013)

### CODE REVIEW

**Comment by tbenz9** [comment]
Wow, that was fast.  Seems a little silly but if this changes the behavior to keep existing contracts active you should probably remove the warning I just put in with #3049.

File: cmd/siac/rentercmd.go [hunk 1]
```diff
@@ -273,7 +273,8 @@ func renterallowancecmd() {
 
 // renterallowancecancelcmd cancels the current allowance.
 func renterallowancecancelcmd() {
-	fmt.Println("Canceling your allowance will destroy all contracts and uploaded files immediately.")
+	fmt.Println(`Canceling your allowance will prevent the renewal of your contracts and you will no longer be able to upload new files.
+	Already uploaded files can still be downloaded until the contracts expire and/or the file's redundancy drops below 1x.`)
```
Comment (lukechampine) [inline cmd/siac/rentercmd.go]: needs to be wrapped to 80 chars

**Comment by DavidVorick** [comment]
We got lucky with the implementation here. If we had made a few different design choices earlier, this would have been more difficult to implement.

As it were, we do try to program such that changing things later is easy, but in this case I think we really did just get lucky.

File: cmd/siac/rentercmd.go (hunk 1 above)
Comment (DavidVorick) [inline cmd/siac/rentercmd.go]: "Canceling your allowance will disable uploading new files, repairing existing files, and renewing existing files. All files will cease to be accessible after a short period of time."

**Comment by DavidVorick** [comment]
Other than the user-facing message, this looks good, including the test.

**Pull request merged**